keras/engine/training.py

```diff
@@ -1458,7 +1458,10 @@ def _standardize_user_data(self, x, y,
         output_shapes = []
         for output_shape, loss_fn in zip(self._feed_output_shapes, self._feed_loss_fns):
             if loss_fn is losses.sparse_categorical_crossentropy:
-                output_shapes.append(output_shape[:-1] + (1,))
+                if K.image_data_format() == 'channels_last':
```

Comment:
Similarly, it seems dangerous to rely on `K.image_data_format` in this case. But since the code is already making assumptions, it's probably okay in this specific case.